logstash-core/lib/logstash/output_delegator_strategy_registry.rb

```diff
@@ -0,0 +1,28 @@
+module LogStash; class OutputDelegatorStrategyRegistry
+  # This is generally used as a singleton
+  # Except perhaps during testing
+  def self.instance
+    @instance ||= self.new
+  end
+
+  def initialize()
+    @map = {}
+  end
+
+  def classes
+    @map.values
+  end
+
+  def types
+    @map.keys
+  end
+  
+  def get(type)
+    @map[type]
```

Comment:
As mentioned above, consider changing this name to `class_for(type)` (its not immediately obvious from the class name `OutputDelegatorStrategyRegistry` that classes are stored within) and raising an error when not found here.